nodes/analyzer/weighted_vector_sum.py

```diff
@@ -0,0 +1,453 @@
+                    b_faces = []
+                    for face in bm_I.faces:
+                        b_faces.append(face)
+                    res = bmesh.ops.triangulate(
+                        bm_I, faces=b_faces, quad_method=self.quad_mode, ngon_method=self.ngon_mode
+                    )
+                    new_vertices_I, new_edges_I, new_faces_I = pydata_from_bmesh(bm_I)
+                    bm_I.free()
+
+                    result_vertices.append(new_vertices_I)
+                    result_edges.append(new_edges_I)
+                    result_polygons.append(new_faces_I)
+                    
+                    verts_I            = np.array(new_vertices_I)
+                    faces_I            = np.array(new_faces_I)
+                    tris_I             = verts_I[faces_I]
+                    areases_I          = np.linalg.norm(np.cross(tris_I[:,1]-tris_I[:,0], tris_I[:,2]-tris_I[:,0]) / 2.0, axis=1)
+                    area_I             = areases_I.sum()
+                    tris_centers_I     = tris_I.sum(axis=1) / 3.0
+                    center_mass_mesh_I = (tris_centers_I * areases_I[np.newaxis].T).sum(axis=0) / area_I
+                    mass_I             = area_I * density_I[0]
+
+                    center_mass_mesh_list.append( center_mass_mesh_I.tolist() )
+                    mass_mesh_list.append(mass_I)
+                    size_mesh_list.append(area_I)
+
+        elif self.center_mode=='VOLUMES':
+            # If density is one list and objects are many then
+            # spead list of density for every mesh. ex.: [[mesh1],[mesh2],[mesh3]], [density:d1,d2,d3] ] => [[mesh1],[mesh2],[mesh3]], [density:[d1],[d2],[d3]] ]
+            if len(input_density_s_2)==1 and len(input_vertices_s_3)>1:
+                input_density_s_2 = [[d] for d in input_density_s_2[0]]
+
+            _, _input_mass_vert_s_2, _input_density_s_2 = match_long_repeat( [input_vertices_s_3, input_mass_vert_s_2[:len(input_vertices_s_3)], input_density_s_2[:len(input_vertices_s_3)]])
+            meshes = match_long_repeat([input_vertices_s_3, input_edges_s_3, input_polygons_s_3, _input_mass_vert_s_2, _input_density_s_2])
+
+            for vertices_I, edges_I, faces_I, mass_of_vertices_I, density_I in zip(*meshes):
+                faces_I_np = np.array(faces_I)
+                if faces_I_np.size==0:
+                    # if no faces at all:
+                    result_mask.append(False)
+                else:
+
+                    # triangulate mesh
+                    bm_I = bmesh_from_pydata(vertices_I, edges_I, faces_I, markup_face_data=True, normal_update=True)
+                    # test if all edges are contiguous (https://docs.blender.org/api/current/bmesh.types.html#bmesh.types.BMEdge.is_contiguous)
+                    # then volume is closed:
+                    for edge in bm_I.edges:
+                        if edge.is_contiguous==False:
```

Comment:
Also probably needs a flag to disable this check. If the user already knows the volume is closed, this is just a waste of time.

Reply:
https://github.com/nortikin/sverchok/pull/5092#discussion_r1536803490